File: java/OOAD/TicTacToe/src/com/techlabs/tictactoe/Cell.java
package com.techlabs.tictactoe;

public class Cell {
	private Mark mark;
	private int rowIndex;
	private int colIndex;
	private int position;

	public Cell(int rowIndex,int colIndex,int position) {
		this.mark = mark.N;
		this.colIndex=colIndex;
		this.rowIndex=rowIndex;
		this.position=position;
	}



	public void setCellMark(Mark mark) {
		try {
			if (this.mark!=Mark.N) {
				throw new Exception("Mark exists At this Place try again");
			} else {
				this.mark = mark;
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	
		
	public int getPostion(){
		return this.position;
	}
	public int  getRowIndex(){
		return this.rowIndex;
	}
	public int getColIndex(){
		return this.colIndex;
	}
	public Mark getCellMark() {
		return this.mark;
	}
	
	public String toString(){
		return mark.toString();
	}

}

File: dotNet/classlibrary/WinForm/TicTacToeApp/TicTacToeCore/Cell.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TicTacToeCore
{
    public class Cell
    {
        private MarkType _mark;
        private int _rowIndex;
        private int _colIndex;
        private int _position;
        
        public Cell(int rowindex,int colindex,int postion)
        {
            this._mark = MarkFactory.NoValue;
            this._rowIndex = rowindex;
            this._colIndex = colindex;
            this._position = postion;
        }

        public MarkType CellMark
        {
            get
            {
                return _mark;
            }
            set
            {
                if (this._mark.Mark != MarkFactory.NoValue.Mark)
                {
                    throw new Exception("Mark exists at this place try again..");
                }
                else
                {
                    this._mark = value;
                }
            }
           
        }
       

        public int RowIndex
        {
            get
            {
                return _rowIndex;
            }
        }

        public int ColIndex
        {
            get
            {
                return _colIndex;
            }
        }

        public int Position
        {
            get
            {
                return _position;
            }
        }

        public override string ToString()
        {
            return _mark.Mark;
        }
    }
}
